src/greynirseq/nicenlp/tasks/document_translation_from_pretrained_bart.py

```diff
@@ -0,0 +1,385 @@
+# Copyright (C) Miðeind ehf.
+# This file is part of GreynirSeq <https://github.com/mideind/GreynirSeq>.
+# See the LICENSE file in the root of the project for terms of use.
+
+# Based on fairseq/tasks/translation.py and fairseq/tasks/translation_from_pretrained_bart.py
+# that has the following license
+#
+#    Copyright (c) Facebook, Inc. and its affiliates.
+#
+#    This source code is licensed under the MIT license found in the
+#    LICENSE file in the root directory of this source tree.
+
+import copy
+import glob
+import logging
+import os
+import pathlib
+from dataclasses import dataclass, field
+from typing import Dict, cast
+
+from fairseq import utils
+from fairseq.data import Dictionary, FairseqDataset, data_utils, iterators
+from fairseq.data.encoders.sentencepiece_bpe import SentencepieceBPE, SentencepieceConfig
+from fairseq.tasks import register_task
+from omegaconf import II
+
+from greynirseq.nicenlp.data.batch_sampler import batch_by_size
+from greynirseq.nicenlp.data.char_noise import CharacterNoiserConfig
+from greynirseq.nicenlp.data.parallel_documents.indexed_parallel_bt_documents_dataset import (
+    IndexedParallelBTDocumentsDataset,
+)
+from greynirseq.nicenlp.data.parallel_documents.indexed_parallel_documents_dataset import (
+    IndexedParallelDocumentsDataset,
+)
+from greynirseq.nicenlp.data.word_noise import WordNoiserConfig
+
+from .translation_from_pretrained_bart import TranslationFromPretrainedBARTConfig, TranslationFromPretrainedBARTTask
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class DocumentTranslationFromPretrainedBARTConfig(TranslationFromPretrainedBARTConfig):
+    max_sequence_length: int = field(
+        default=1000,
+        metadata={"help": "Maximum sequence length to train model on. Should be less than the positional encoding."},
+    )
+    num_preprocess_workers: int = field(
+        default=2,
+        metadata={"help": "number of workers to preprocess the data"},
+    )
+    bt_subset: str = field(
+        default="",
+        metadata={
+            "help": "Prefix of backtranslation included in the training data.\
+Should include the 'train_subset' prefix, i.e. if train_subset='train' then bt_subset='train_bt' will work."
+        },
+    )
+    parallel_prob: float = field(
+        default=0.33,
+        metadata={"help": "Probability of sampling parallel data if bt data is included (Note: NOT sample weight)"},
+    )
+    fragment_noise_prob: float = field(
+        default=0.01,
+        metadata={"help": "Probability of fragment noise"},
+    )
+    max_merges: int = field(
+        default=10,
+        metadata={"help": "How many segments are at most merged into a single training example."},
+    )
+    global_skip_noise_prob: float = field(
```

Comment:
What happens to the skipped segment's neighbours? Do they become "neighbours" of each other?